Sources/MongoSwift/BSON/Overwritable.swift

```diff
@@ -7,8 +7,8 @@ internal protocol Overwritable: BSONValue {
      * Overwrites the value at the current position of the iterator with self.
      *
      * - Throws:
-     *   - `MongoError.InternalError` if the `BSONValue` is an `Int` and cannot be written to BSON.
-     *   - `MongoError.LogicError` if the `BSONValue` is a `BSONDecimal128` or `BSONObjectID` and is improperly
+     *   - `BSONError.InternalError` if the `BSONValue` is an `Int` and cannot be written to BSON.
```

Comment:
these two can go away as well. in fact I think we could just make the protocol stop throwing